utoipa-swagger-ui/src/lib.rs

```diff
@@ -481,6 +517,7 @@ impl<'a> Config<'a> {
     pub fn new<I: IntoIterator<Item = U>, U: Into<Url<'a>>>(urls: I) -> Self {
         Self {
             urls: urls.into_iter().map(|url| url.into()).collect(),
+            oauth: None,
         }
     }
```

Comment:
I would add here a method `with_oauth_config` something like this.
```rust
pub fn with_oauth_config<I: IntoIterator<Item = U>, U: Into<Url<'a>>>(urls: I, 
   oauth_config: oauth::Config) -> Self {
        Self {
            urls: urls.into_iter().map(|url| url.into()).collect(),
            oauth: Some(oauth_config),
        }
}
```
This `Config` type is `non_exhaustive` for backwards compatibility purposes

>Outside of the defining crate, types annotated with non_exhaustive have limitations that preserve backwards compatibility when new fields or variants are added.
>
>Non-exhaustive types cannot be constructed outside of the defining crate:
>
>    Non-exhaustive variants ([struct](https://doc.rust-lang.org/reference/items/structs.html) or enum variant) cannot be constructed with a StructExpression (including with functional update syntax).
>    enum instances can be constructed.
https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute

It is in Rust a good idea to define outside looking API with `non_exhaustive` and instead provide constructors `new()` or / and accessor fields or builder pattern to construct objects. This way the API implementator is at control and not every change to the API nor adding / changing / deleting one field will be a breaking change for API consumer.